model/tasks/ChangePermissionsTask.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoDacSimple\model\tasks;
+
+use common_exception_MissingParameter;
+use common_report_Report as Report;
+use Exception;
+use JsonSerializable;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\extension\AbstractAction;
+use oat\oatbox\service\ServiceManagerAwareTrait;
+use oat\tao\model\taskQueue\Task\TaskAwareInterface;
+use oat\tao\model\taskQueue\Task\TaskAwareTrait;
+use oat\taoDacSimple\model\PermissionsService;
+use oat\taoDacSimple\model\PermissionsServiceFactory;
+
+
+/**
+ * Class ChangePermissionsTask
+ *
+ * Handling permission changes in background
+ */
+class ChangePermissionsTask extends AbstractAction implements TaskAwareInterface, JsonSerializable
+{
+    use ServiceManagerAwareTrait;
+    use TaskAwareTrait;
+    use OntologyAwareTrait;
+
+    public const PARAM_RECURSIVE = 'recursive';
+    public const PARAM_RESOURCE = 'resource';
+    public const PARAM_PRIVILEGES = 'privileges';
+
+    public function __invoke($params = []): Report
+    {
+        $this->validateParams($params);
+        try {
+            $service = $this->getPermissionService();
+            $service->savePermissions(
+                (bool)$params[self::PARAM_RECURSIVE],
+                $this->getClass($params[self::PARAM_RESOURCE]),
+                $params[self::PARAM_PRIVILEGES]
+            );
+            $result = Report::createSuccess('Permission applied');
```

Comment:
We should have the same message on Task Queue report as on instant feedback.
What about changing this to `Permissions saved`?

Reply:
Changed